contracts/core/OptionsPremiumPricer.sol

```diff
@@ -0,0 +1,227 @@
+    IPriceOracle public immutable priceOracle;
+    ICompToken public immutable cToken;
+
+    // For reference - IKEEP3rVolatility: 0xCCdfCB72753CfD55C5afF5d98eA5f9C43be9659d
+
+    constructor(
+        address _volatilityOracle,
+        address _priceOracle,
+        address _cToken
+    ) {
+        require(_volatilityOracle != address(0), "!_volatilityOracle");
+        require(_priceOracle != address(0), "!_priceOracle");
+        require(_cToken != address(0), "!_cToken");
+        volatilityOracle = IVolatilityOracle(_volatilityOracle);
+        priceOracle = IPriceOracle(_priceOracle);
+        cToken = ICompToken(_cToken);
+    }
+
+    /**
+     * @notice Calculates the premium of the provided option using Black-Scholes
+     * References for Black-Scholes:
+       https://www.macroption.com/black-scholes-formula/
+       https://www.investopedia.com/terms/b/blackscholes.asp
+       https://www.erieri.com/blackscholes
+       https://goodcalculators.com/black-scholes-calculator/
+       https://www.calkoo.com/en/black-scholes-option-pricing-model
+     * @param st is the strike price of the option
+     * @param expiryTimestamp is the unix timestamp of expiry
+     * @param isPut is whether the option is a put option
+     * @return premium for 100 contracts with 8 decimals i.e. 500*10**8 = $500 for 100 contracts
+     */
+    function getPremium(
+        uint256 st,
+        uint256 expiryTimestamp,
+        bool isPut
+    ) external view returns (uint256 premium) {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        (uint256 sp, uint256 v, uint256 t) =
+            blackScholesParams(expiryTimestamp);
+
+        (uint256 call, uint256 put) = quoteAll(t, v, sp, st);
+
+        premium = !isPut ? call : put;
+    }
+
+    /**
+     * @notice Calculates the option's delta
+     * Formula reference: `d_1` in https://www.investopedia.com/terms/b/blackscholes.asp
+     * http://www.optiontradingpedia.com/options_delta.htm
+     * https://www.macroption.com/black-scholes-formula/
+     * @param st is the strike price of the option
+     * @param expiryTimestamp is the unix timestamp of expiry
+     * @return delta for given option. 2 decimals (ex: 81 = 0.81 delta) as this is what strike selection
+     * module recognizes
+     */
+    function getOptionDelta(uint256 st, uint256 expiryTimestamp)
+        external
+        view
+        returns (uint256 delta)
+    {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        (uint256 sp, uint256 v, uint256 t) =
+            blackScholesParams(expiryTimestamp);
+
+        uint256 d1;
+        uint256 d2;
+
+        // Divide delta by 10 ** 12 to bring it to 2 decimals for strike selection
+        if (sp >= st) {
+            (d1, d2) = d(t, v, sp, st);
+            delta = Math.ncdf((Math.FIXED_1 * d1) / 1e18).div(10**12);
```

Comment:
dividing by 10 ** 12 here because strike selection module expects 2 decimals ( ex: delta 1 is 100, delta 0.89 is 89)